pkg/agent/cniserver/interface_configuration_windows.go

```diff
@@ -99,10 +100,19 @@ func (ic *ifConfigurator) configureContainerLink(
 	mtu int,
 	result *current.Result,
 ) error {
-	// Create HNS Endpoint.
-	endpoint, err := ic.createContainerLink(podName, podNameSpace, containerID, result)
-	if err != nil {
-		return err
+	epName := util.GenerateContainerInterfaceName(podName, podNameSpace)
+	// Search endpoint from local cache.
+	endpoint, found := ic.getEndpoint(epName)
+	if !found {
+		if !ipam.IsInfraContainer(containerNetNS) {
```

Comment:
Feel IsInfraContainer is specific to ipam

Reply:
Yeah, it's little odd to place `IsInfraContainer` in ipam.  I move it to agent.util.net_windows.go/net_linux.go in update.

Reply:
Feel util is not ideal for it too, could it just under pkg cniserver as it's a concept of CNI request?